robot/packages/cloud_package_manager.go

```diff
@@ -173,8 +174,19 @@ func (m *cloudManager) Sync(ctx context.Context, packages []config.PackageConfig
 
 		m.logger.Debugf("Downloading from %s", sanitizeURLForLogs(resp.Package.Url))
 
+		nonEmptyPaths := make([]string, 0)
+		if p.Type == config.PackageTypeModule {
+			matchedModules := config.ModulesForPackage(p.Name, modules)
+			if len(matchedModules) == 1 {
+				nonEmptyPaths = append(nonEmptyPaths, config.RemovePlaceholderPrefix(matchedModules[0].ExePath))
+			}
+			if len(matchedModules) > 1 {
+				m.logger.Warnf("package %s matched %d > 1 modules, not doing entrypoint checking", p.Name, len(matchedModules))
+			}
```

Comment:
```suggestion
        if p.Type == config.PackageTypeModule {
            for _, possibleModule := range modules {
                if strings.HasPrefix(possibleModule.ExePath, p.LocalDataDirectory(m.packagesDir)){
                    nonEmptyPaths = append(nonEmptyPaths, possibleModule.ExePath)
                }
            }
        }

```

(I'd advise to just store the whole abs path, but if you want to keep your relative path logic, you could trim the prefix)

Reply:
(you could also just keep passing the modules down to the downloadPackage -> checkNonemptyPaths fn and do the string prefix checking there)